extension/chrome/settings/modules/contacts.ts

```diff
@@ -23,6 +33,48 @@ Catch.try(async () => {
 
   BrowserMsg.listen(tabId); // set_css
 
+  const attUI = new AttUI(() => Promise.resolve({ size_mb: 5, size: 5 * 1024 * 1024, count: 1 }));
+  attUI.initAttDialog('fineuploader', 'fineuploader_button');
+  attUI.setAttAddedCb(async (file: Att) => {
+    (file !== null && typeof file !== 'undefined' && typeof file.id !== 'undefined' && file.id !== '') ? attUI.clearAtt(file.id) : attUI.clearAtts();
+    await handleKeyfileSelected(file);
+  });
+
+  const handleKeyfileSelected = async (file: Att) => {
+    try {
+      const rawFile = file.getData();
+      const stringFile = new TextDecoder("utf-8").decode(rawFile);
+      const { blocks } = await Pgp.armor.detectBlocks(stringFile);
+
+      const keys: OpenPGP.key.Key[] = [];
+      if (blocks !== null && blocks.length > 0 && blocks.some(block => block.type === "publicKey")) {
+        for (const block of blocks) {
+          if (block.type === "publicKey") {
+            const result = await openpgp.key.readArmored(block.content);
+            if (result.err !== null && typeof result.err !== 'undefined') {
```

Comment:
can simplify to `if (result.err)`.

However, I think it would be better to ignore these errors. If at least one pubkey can be processed, I'd consider that a success.